roles/rsyslog/templates/input_remote.j2

```diff
@@ -0,0 +1,46 @@
+{% if remote_item.udp_port | d() %}
+# Log messages from remote hosts over UDP
+input(name="{{ remote_item.name }}" type="imudp" port="{{ remote_item.udp_port | d(__rsyslog_default_port) }}")
+{% elif remote_item.tcp_port | d() %}
+# Log messages from remote hosts over plain TCP
+input(name="{{ remote_item.name }}" type="imptcp" port="{{ remote_item.tcp_port | d(__rsyslog_default_port) }}")
+{% else %}
+# Log messages from remote hosts over TLS
+input(name="{{ remote_item.name }}" type="imtcp" port="{{ remote_item.tcp_tls_port | d(__rsyslog_default_tls_port) }}")
+{% endif %}
```

Comment:
I think you can use `{{ lookup('template', 'input_template.j2') }}` here to avoid having to copy/paste the contents of input_template.j2 here.  In that sense, yes, you could think of a template as a "library" of "functions" that you can "include" in multiple places.

https://docs.ansible.com/ansible/latest/plugins/lookup/template.html

Reply:
Ah, I see. We can template in a template file... :O

Reply:
Or inherit from templates, not sure whether it helps here though: https://jinja.palletsprojects.com/en/2.9.x/templates/#template-inheritance